evalml/tests/utils_tests/test_woodwork_utils.py

```diff
@@ -252,3 +256,54 @@ def test_infer_feature_types_NA_to_nan(null_col, already_inited):
         assert all([isinstance(x, type(np.nan)) for x in inferred_df["unknown"]])
     else:
         assert all([isinstance(x, str) for x in df["unknown"]])
+
+
+@pytest.mark.parametrize(
+    "logical_types,l_equal",
+    [
+        ({"first": Categorical(), "second": Integer(), "third": Double()}, True),
+        ({"first": Categorical(), "second": Double(), "third": Double()}, True),
+        ({"first": Categorical(), "second": Double(), "third": Categorical()}, False),
+    ],
+)
+@pytest.mark.parametrize(
```

Comment:
Can we add some cases with multiple semantic tags and some of the less common logical types such as Email/URL/DateTime/Unknown?